src/dvclive/monitor_system.py

```diff
@@ -0,0 +1,158 @@
+import logging
+from typing import Dict, Union, Optional, List, Tuple
+from pathlib import Path
+
+import psutil
+from statistics import mean
+from threading import Event, Thread
+from funcy import merge_with
+
+from .error import InvalidDataTypeError
+
+
+logger = logging.getLogger("dvclive")
+MEGABYTES_DIVIDER = 1024.0**2
+GIGABYTES_DIVIDER = 1024.0**3
+
+MINIMUM_CPU_USAGE_TO_BE_ACTIVE = 20
+
+
+class _MonitorSystem:
```

Comment:
it it abstract class, consider using ABC (I'm not an expert in those, but it seems we should be using it here cc @skshetry )